packages/core-data/src/actions.js

```diff
@@ -946,8 +946,10 @@ export function receiveRevisions(
 	invalidateCache = false,
 	meta
 ) {
+	const isTemplate = [ 'wp_template', 'wp_template_part' ].includes( name );
 	return {
 		type: 'RECEIVE_ITEM_REVISIONS',
+		key: isTemplate ? 'wp_id' : DEFAULT_ENTITY_KEY,
```

Comment:
Why do we have a special case for templates?

Reply:
Good question. 

Template records  have ids that look like this: `theme_name//template_slug`. 

This is fine for storing template entities themselves, since the ids are unique are usually unique:

```js
{
    'theme_name//template_slug_1': { ...data },
    'theme_name//template_slug_2': { ...data },
}
```

These template records have a property `wp_id`, whose value is the **real** post id (the unique key in the database). 

The spicy bit is that templates revisions use the same the schema as their parents:

```js
// a bunch of template revisions
[
    {
        id: 'theme_name//template_slug_1',
        wp_id: 2 // revision id
    },
    {
        id: 'theme_name//template_slug_1',
        wp_id: 3 // revision id
    },
]
```

So we can't store them by the default key `id` as we do with other post types since it's not unique. We have to use `wp_id`.

What I've done is temporary, and I think the key should be defined somewhere in `entityConfig` - not sure how to do it cleanly yet as it's only required for revisions.

Reply:
AFAIK, the "id" is unique (I implemented that system)

Reply:
> the "id" is unique 

For the entities or their revision?

Here's the raw output of `/wp/v2/templates/twentytwentythree/wp-custom-template-roo/revisions`, where `wp-custom-template-roo` is a template I created with some changes.

<details>

<summary>API response</summary>

```js
[
    {
        "id": "twentytwentythree//wp-custom-template-roo",
        "theme": "twentytwentythree",
        "content": {
            "raw": "<!-- wp:paragraph -->\n<p>roo12</p>\n<!-- /wp:paragraph -->"
        },
        "slug": "142-revision-v1",
        "source": "custom",
        "origin": null,
        "type": "revision",
        "description": "",
        "title": {
            "raw": "roo",
            "rendered": "roo"
        },
        "status": "inherit",
        "wp_id": 145,
        "has_theme_file": false,
        "author": 1,
        "modified": "2023-11-22T04:02:31",
        "parent": 142,
        "_links": {
            "self": [
                {
                    "href": "http://localhost:8888/index.php?rest_route=/wp/v2/templates/twentytwentythree//wp-custom-template-roo/revisions/145"
                }
            ],
            "parent": [
                {
                    "href": "http://localhost:8888/index.php?rest_route=/wp/v2/templates/twentytwentythree//wp-custom-template-roo"
                }
            ]
        }
    },
    {
        "id": "twentytwentythree//wp-custom-template-roo",
        "theme": "twentytwentythree",
        "content": {
            "raw": "<!-- wp:paragraph -->\n<p>roo1</p>\n<!-- /wp:paragraph -->"
        },
        "slug": "142-revision-v1",
        "source": "custom",
        "origin": null,
        "type": "revision",
        "description": "",
        "title": {
            "raw": "roo",
            "rendered": "roo"
        },
        "status": "inherit",
        "wp_id": 144,
        "has_theme_file": false,
        "author": 1,
        "modified": "2023-11-22T04:02:28",
        "parent": 142,
        "_links": {
            "self": [
                {
                    "href": "http://localhost:8888/index.php?rest_route=/wp/v2/templates/twentytwentythree//wp-custom-template-roo/revisions/144"
                }
            ],
            "parent": [
                {
                    "href": "http://localhost:8888/index.php?rest_route=/wp/v2/templates/twentytwentythree//wp-custom-template-roo"
                }
            ]
        }
    },
    {
        "id": "twentytwentythree//wp-custom-template-roo",
        "theme": "twentytwentythree",
        "content": {
            "raw": "<!-- wp:paragraph -->\n<p>roo</p>\n<!-- /wp:paragraph -->"
        },
        "slug": "142-revision-v1",
        "source": "custom",
        "origin": null,
        "type": "revision",
        "description": "",
        "title": {
            "raw": "roo",
            "rendered": "roo"
        },
        "status": "inherit",
        "wp_id": 143,
        "has_theme_file": false,
        "author": 1,
        "modified": "2023-11-22T04:02:26",
        "parent": 142,
        "_links": {
            "self": [
                {
                    "href": "http://localhost:8888/index.php?rest_route=/wp/v2/templates/twentytwentythree//wp-custom-template-roo/revisions/143"
                }
            ],
            "parent": [
                {
                    "href": "http://localhost:8888/index.php?rest_route=/wp/v2/templates/twentytwentythree//wp-custom-template-roo"
                }
            ]
        }
    }
]
```

</details>

Reply:
For the entities, the "templates" are not always in the database so wp_id is not always set (it's not set if you load a raw untouched theme template). By yes, revisions are only for the "saved" templates, so the REST API for revisions is probably using wp_id.

The same should apply to template parts.

So yeah, there's a special case here for sure, I think it's probably ok to generate the URL to fetch revisions using `wp_id` for templates and template parts but it seems all the rest (all the places in our reducers....) should continue using id.